### ISSUE TITLE
[Bug] Starting with_rosmaster=true and then publishing causes crash

### ISSUE THREAD

**Issue by bearrito** [eclipse-zenoh/zenoh-plugin-ros1#9]
### Describe the bug

Related to my other ticket #8 

I can start the bridge with

```
locus@3479b00d30b9:~/src/zenoh-plugin-ros1$ ./target/release/zenoh-bridge-ros1 --with_rosmaster true --ros_master_uri http://localhost:11311
[2023-08-18T16:35:33Z INFO  zenoh_bridge_ros1] zenoh-bridge-ros1 v0.10.0-dev-3-g51bf400 built with rustc 1.71.0 (8ede3aae2 2023-07-12)
[2023-08-18T16:35:33Z INFO  zenoh::net::runtime] Using PID: c6ad0ccfdaac4f1828d8ffe8e8d1cacd
[2023-08-18T16:35:33Z INFO  zenoh::net::runtime::orchestrator] Zenoh can be reached at: tcp/172.17.0.2:34877
[2023-08-18T16:35:33Z INFO  zenoh::net::runtime::orchestrator] zenohd listening scout messages on 224.0.0.224:7446

# At this point I start publishing in another terminal

thread 'async-std/runtime' panicked at 'called `Result::unwrap()` on an `Err` value: PatternNotMatched', zenoh-plugin-ros1/src/ros_to_zenoh_bridge/discovery.rs:194:10
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
Aborted (core dumped)
```

Then in another window I do

```
locus@3479b00d30b9:/$ rostopic pub /foo std_msgs/String "data: 'Hello'" 
publishing and latching message. Press ctrl-C to terminate
```

At that point the bridge will crash.

### To reproduce

![crashes_at_publish](https://github.com/eclipse-zenoh/zenoh-plugin-ros1/assets/1908139/c7a7f54e-a949-49c0-b310-f2adb6d8c96a)

### System info

Ubuntu 20 Workstation and Ubuntu 20 Docker Image
commit 51bf400283f39ea554028b1499da876288b608e3

**Comment by JEnoch** [comment]
The problem comes from the data type which is `std_msgs/String`.
The current design of the bridge tries and fails to convert this string as a key expression segment where `/` is not allowed.
See the expected keyexpr format here: https://github.com/eclipse-zenoh/zenoh-plugin-ros1/blob/c04fd0d6de704650cdb6263c18fa3a45273a0f27/zenoh-plugin-ros1/src/ros_to_zenoh_bridge/discovery.rs#L32

@yellowhatter : I guess your code won't support changing the format with `${data_type:**}`, right?
Parsing back with such pattern might lead to confusion between `data_type`, `bridge_namespace` and `topic`.
A solution could be to replace in `data_type` all `/` with `.` (or any other escape character) before insertion in keyexpr. And replacing back where appropriate.
Any other suggestion ?

**Comment by yellowhatter** [comment]
@JEnoch  correct, the problem is in unexpected '/' in the 'data_type' field. Any escaping is impossible because formally 'data_type' seems to support any characters at all and thus we cannot escape correctly. I would rather try encoding 'data_type' as hexadecimal byte sequence - that will give us suitable strings like 'deadbeeef00' to place into zenoh key.

**Comment by JEnoch** [comment]
I don't think that `data_type` can contain any character.

As far as I understand, this `Topic.data_type` contains the ROS topic's message type, that is specified [here](https://wiki.ros.org/msg) 
 > _"Message types are referred to using [package resource names](https://wiki.ros.org/Names#Package_Resource_Names)"_ 

And valid package resource name can contain only alphanumeric characters, `_` or `/` (at most one).
[This code](https://github.com/adnanademovic/rosrust/blob/c46a7e6742e5e01e0271d144ae2a05be9a035539/ros_message/src/data_type.rs#L185-L195) parsing datatype in rosrust seems to confirm that.

Is such "ros1_discovery_info" key expr parsed only [here](https://github.com/eclipse-zenoh/zenoh-plugin-ros1/blob/c04fd0d6de704650cdb6263c18fa3a45273a0f27/zenoh-plugin-ros1/src/ros_to_zenoh_bridge/discovery.rs#L136), or do you know any other place where the data_type could be extracted from it ?

**Comment by yellowhatter** [comment]
@JEnoch this is the only place. This (discovery) is a code that is planned to be completely changed at the bridge's release. Anyway, it is an "internals" of the bridge (and the keyspace used for discovery is internal too) thus we are free to change it at any time.

I agree that we can rely on data_type conventions, but nobody prevents rosmaster environment from an ill-formed ros1 client implementation - we should try to be stable where possible and shouldn't fail in situation where we don't need to fail.
I've prepared a [PR](https://github.com/eclipse-zenoh/zenoh-plugin-ros1/pull/11) that fixes this issue and adds tests for it.
